### PR TITLE
Allow Disabling Storage Labels

### LINKED ISSUE #3242: Start a Local Agent without Default Storage Labels
## Current behavior

Starting a LocalAgent will add storage labels by default, which can be confusing for new users and cluttered for users with many agents. Typically, starting a LocalAgent will produce the following by default:

`[2020-09-01 15:34:50,612] INFO - agent | Starting LocalAgent with labels ['my-machine.local', 'azure-flow-storage', 'gcs-flow-storage', 's3-flow-storage', 'github-flow-storage', 'webhook-flow-storage']`

## Proposed behavior

Starting a Local Agent should automatically exclude the following labels: 
'azure-flow-storage', 'gcs-flow-storage', 's3-flow-storage', 'github-flow-storage', 'webhook-flow-storage'

These labels should have some option to turn off by default or be removed by default.

Implemented here: https://github.com/PrefectHQ/prefect/blob/5edb9b39bb53e2cb11d37832f3f3f56bb1a3bc6a/src/prefect/agent/local/agent.py#L97

## Example

prefect agent start -t "TOKEN"
Produces:
```
[2020-09-01 17:56:00,216] INFO - agent | Starting LocalAgent with labels ['my-machine.local', 'azure-flow-storage', 'gcs-flow-storage', 's3-flow-storage', 'github-flow-storage', 'webhook-flow-storage']
[2020-09-01 17:56:00,216] INFO - agent | Agent documentation can be found at https://docs.prefect.io/orchestration/
[2020-09-01 17:56:00,216] INFO - agent | Agent connecting to the Prefect API at https://api.prefect.io
[2020-09-01 17:56:00,532] INFO - agent | Waiting for flow runs...
```
But can clutter the UI:

Potential Results:
Agent management will be decluttered for users with many Agents.
Quality of life change - specifying execution environments via flow affinity is a bit easier to navigate.

Please let me know if there is a current solution or reasoning I'm not aware of!

### PR DESCRIPTION
**Pull request by raphael-riel**
## Summary
For the neat-freak like me, all those storage labels meant nothing.
Here's an option to disable their addition.
Fixes #3242

See conversation: https://prefect-community.slack.com/archives/CL09KU1K7/p1602705100472300

## Changes
- Add option for LocalAgent to NOT add storage labels.
- Same option for the `agent start` CLI.
- Bonus: Added the CLI option for not adding hostname label.

All default behaviours are preserved.

## Checklist

This PR:

- [x] adds new tests (if appropriate)
- [x] adds a change file in the `changes/` directory (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

### CODE REVIEW

File: src/prefect/cli/agent.py [hunk 1]
```diff
@@ -208,6 +224,8 @@ def start(
                                         e.g. `-p /root/my_scripts -p /utilities`
         --show-flow-logs, -f        Display logging output from flows run by the agent
                                     (available for Local and Docker agents only)
+        --hostname-label            Add hostname to the Agent's labels
+        --storage-labels            Add all available storage label to the Agent
```
Comment (joshmeek) [inline src/prefect/cli/agent.py:228]: Since both of these default to true we should probably document here that the `--no-XXXX` option will disable it

**Comment by joshmeek** [comment]
This looks great @raphael-riel! One minor docstring comment and then I think with the addition of the `/changes` entry this is good to go

File: src/prefect/cli/agent.py (hunk 1 above)
Comment (raphael-riel) [reply src/prefect/cli/agent.py]: Added!

**Comment by raphael-riel** [comment]
@joshmeek All good now!

**Comment by raphael-riel** [comment]
Proof included!
image
image

File: src/prefect/cli/agent.py [hunk 2]
```diff
@@ -208,6 +224,10 @@ def start(
                                         e.g. `-p /root/my_scripts -p /utilities`
         --show-flow-logs, -f        Display logging output from flows run by the agent
                                     (available for Local and Docker agents only)
+        --hostname-label            Add hostname to the Agent's labels
+                                        (Default to True. Disable with --no-hostname-label option)
+        --storage-labels            Add all available storage label to the Agent
```
Comment (madkinsz) [inline src/prefect/cli/agent.py:229]:
```suggestion
        --storage-labels            Add all available storage labels to the Agent
```

File: src/prefect/cli/agent.py [hunk 3]
```diff
@@ -69,6 +69,20 @@ def agent():
     help="Labels the agent will use to query for flow runs.",
     hidden=True,
 )
+@click.option(
+    "--storage-labels/--no-storage-labels",
+    type=bool,
+    default=True,
+    help="Add all available storage label to the LocalAgent",
```
Comment (madkinsz) [inline src/prefect/cli/agent.py:76]:
```suggestion
    help="Add all available storage labels to the LocalAgent",
```

File: src/prefect/agent/local/agent.py [hunk 4]
```diff
@@ -55,6 +55,8 @@ class LocalAgent(Agent):
         - hostname_label (boolean, optional): a boolean specifying whether this agent should
             auto-label itself with the hostname of the machine it is running on.  Useful for
             flows which are stored on the local filesystem.
+        - storage_labels (boolean, optional): a boolean specifying whether this agent should
+            auto-label itself with all of the available storage options labels.
```
Comment (madkinsz) [inline src/prefect/agent/local/agent.py:59]: minor nit, does it make sense to have "available" in this description (and in similar below)? It seems to imply that we are doing more than attaching all of the possible storage option labels.

**Review by madkinsz — COMMENTED** [review]
Thanks for your contribution! Some minor documentation comments, LGTM otherwise

File: src/prefect/agent/local/agent.py (hunk 4 above)
Comment (raphael-riel) [reply src/prefect/agent/local/agent.py]: Works for me! Fixed in 5bafac89f2efb74a5894ec79905c1a7a72354dba

**Comment by raphael-riel** [comment]
@madkinsz Documentation strings are fixed. Thanks for the input!

**Review by joshmeek — DISMISSED** [review]
🚀

File: changes/pr3503.yaml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+enhancement:
+  - "Allow Disabling Storage Labels - [#3503](https://github.com/PrefectHQ/prefect/pull/3503)"
```
Comment (joshmeek) [inline changes/pr3503.yaml:2]:
```suggestion
  - "Allow disabling default storage labels for the `LocalAgent` - [#3503](https://github.com/PrefectHQ/prefect/pull/3503)"
```

Comment (joshmeek) [reply changes/pr3503.yaml:2]: One minor changes entry nit 🙂

**Review by joshmeek — DISMISSED** [review]

**Review by madkinsz — DISMISSED** [review]

**Comment by AdeSantosSierra** [comment]
Lovely!! Thanks!

**Comment by joshmeek** [comment]
Closing and reopening in an attempt to get the github checks to pick up 🤔

**Comment by raphael-riel** [comment]
Can someone give me a hint about what's going on with CircleCI failing test?

File: src/prefect/cli/agent.py [hunk 6]
```diff
@@ -69,6 +69,20 @@ def agent():
     help="Labels the agent will use to query for flow runs.",
     hidden=True,
 )
+@click.option(
+    "--storage-labels/--no-storage-labels",
+    type=bool,
```
Comment (joshmeek) [inline src/prefect/cli/agent.py:74]:
```suggestion
```

File: src/prefect/cli/agent.py [hunk 7]
```diff
@@ -69,6 +69,20 @@ def agent():
...
+    default=True,
+    help="Add all storage labels to the LocalAgent",
+    hidden=True,
+)
+@click.option(
+    "--hostname-label/--no-hostname-label",
+    type=bool,
```
Comment (joshmeek) [inline src/prefect/cli/agent.py:81]:
```suggestion
```

**Review by joshmeek — COMMENTED** [review]
This is failing on the click lower bound of 7.0.0 and I don't think the bool type is needed

**Pull request merged**